gen/build_deploy/bash.py

```diff
@@ -17,6 +18,21 @@
 from pkgpanda.util import logger
 
 
+def calculate_fault_domain_detect_contents(fault_domain_detect_filename):
+    if not os.path.exists(fault_domain_detect_filename):
```

Comment:
i'd probably better prefer open, at least we know if we have permissions to read the file

Reply:
Either of those approaches will raise an unhandled exception if we don't have permission to read the file, since we'll try to open it at some point regardless as long as the file is there, and `FileNotFoundError` won't catch permissions exceptions. Which is probably a good thing -- if the file exists but the installer can't read it then we should fail immediately rather than silently disable fault domains, since that's probably not the user's intention.